### DISCUSSION TITLE
Don't understand benefit of using getserversideprops instead of getInitialProps

### DISCUSSION THREAD

**Discussion by rajatdhoot123** [vercel/next.js#14066] [Help]
I am Upgrading from getInitialProps -> getServersideprops and saw the API request is made from node server and passes the data the .json API to the client (Client-side routing)

Don't it increases the overhead rather than making the API from the browser and getting response directly from the API now it has to call from node server and getting the response and passing to the client through .json API.

Can Someone please clarify my understanding regarding new update.

**Comment by illuminist** [comment]
`getInitialProps` run first time on serverside, then in sub sequence page routing, `getInitialProps` will run in clientside.

`getServerSideProps` always runs on serverside. In sub sequence page routing, client will send a request to server to run only `getServerSideProps` function to get only prop object and send back to client. As it runs only on server, this part code will never be in clientside which you can write a code that communicates with database directly by using this function.

**Reply by rajatdhoot123** [reply]
So what I understand if we are using a database with node server then only it will benefit.
In the case when we are using API call to get data then it was an extra overhead communication because it first has to make a request to node server then it will make an API call when it gets resolved node server sends data back to the client.
In this case, it increases one level of communication.

**Reply by illuminist** [reply]
`getInitialProps` isn't like going to be deprecated soon. I still have plenty of usecases of it.

`getServerSideProps` acts like combination of page generation and API. So if there already are API endpoints for client to call, then using `getInitialProps` is still more reasonable. But it's available as an option so that's nice too.

**Reply by rajatdhoot123** [reply]
Then in docs, it should not recommend using `getServerSideProps`  instead of `getInitialProps ` if there are use case which was not great fit for `getServerSideProps`

**Comment by zimv** [comment]
The question is my question too.
I think the NextJs document only recommend that using getServerSideProps instead of getInitialProps, but not to remove it in future.
It has  more choice for different application scenarios. So you could use getInitialProps if you want.

**Comment by sromexs** [comment]
i have this issue too
in first requesting to page i want to get data from api
but in second navigation with next/link in client side i dont want to wait to page render in server side but with getServerSideProps  it will be render in server every time
now one ever find solution for this ?

**Reply by illuminist** [reply]
That's the point of using `getServerSideProps`-- to get data from server and that function can be server-only code. Use `getInitialProps` instead if you already have api to fetch data.
